### DISCUSSION TITLE
Error whilst trying to import stores in Snowpack

### DISCUSSION THREAD

**Discussion by DavidEaston** [solidjs/solid#596] [Q&A] [answered]
Hello, sorry if this isn't a great place to ask - I'm really enjoying using Solid-js with a new meta framework, Astro. But one problem I ran into was trying to import a store like this: 
`
import { createStore } from "solid-js/store/dist/store.js";
`

I run into this error: 
`Error: Package "solid-js" exists but package.json "exports" does not include entry for "./dist/server.js/store/dist/store.js".
Did you mean "./dist/server.js"?`

[Snowpack documentation](https://www.snowpack.dev/reference/common-error-details) suggests that this is a common problem and it seems like the only solution is to ask the package authors. If there is anything I can do to assist in resolving this I am happy to help if possible.

**Comment by stephenhutchings** [comment, marked as answer]
Bumping my head against this as well. It looks like the [solid renderer](https://github.com/snowpackjs/astro/blob/main/packages/renderers/renderer-solid/index.js) needs to be updated to allow access to Solid's store exports.

Changing that index file to include the store in `knownEntrypoints`, `external` and `alias` seems to work.

```javascript
export default {
  name: '@astrojs/renderer-solid',
  client: './client',
  server: './server',
  knownEntrypoints: ['solid-js', 'solid-js/web', 'solid-js/store'],
  external: ['solid-js/store/dist/server.js', 'solid-js/web/dist/server.js', 'solid-js/dist/server.js', 'babel-preset-solid'],
  jsxImportSource: 'solid-js',
  jsxTransformOptions: async ({ isSSR }) => {
    const [{ default: solid }] = await Promise.all([import('babel-preset-solid')]);
    const options = {
      presets: [solid({}, { generate: isSSR ? 'ssr' : 'dom' })],
    };

    if (isSSR) {
      options.alias = {
        'solid-js/store': 'solid-js/store/dist/server.js',
        'solid-js/web': 'solid-js/web/dist/server.js',
        'solid-js': 'solid-js/dist/server.js',
      };
    }

    return options;
  },
};
```
That being said, I'm not really sure why this is all necessary as the package exports in Solid look to be configured correctly, but I'm new to Snowpack so ¯\\_ (ツ)_/¯.

**Reply by DavidEaston** [reply]
Thanks a lot for sharing! It might potentially be an issue on snowpacks side, I might raise this on the Snowpack repo and see if we can get any closer to resolving the core issue. 

For now this solution works for me in the mean time - thanks again!
